src/controller/stream-controller.ts

```diff
@@ -1113,15 +1126,37 @@ export default class StreamController
       const details = this.getLevelDetails();
       const buffered = BufferHelper.getBuffered(media);
       const bufferStart = buffered.length ? buffered.start(0) : 0;
-      const delta = bufferStart - startPosition;
+      const mainFrag = this.getBufferedFrag(
+        bufferStart + this.config.maxFragLookUpTolerance,
+        PlaylistLevelType.MAIN,
+        false,
+      );
+      const audioFrag = this.getBufferedFrag(
+        bufferStart + this.config.maxFragLookUpTolerance,
+        PlaylistLevelType.AUDIO,
+        false,
+      );
+      if (
+        (!mainFrag && seekToSegmentBoundary) ||
+        (!audioFrag && startOnAVSegmentBoundary)
+      ) {
+        return false;
+      }
+      const newStartPosition = Math.max(
+        bufferStart,
+        mainFrag?.start ?? 0,
+        audioFrag?.start ?? 0,
+      );
+      const delta = newStartPosition - startPosition;
       const skipTolerance = Math.max(
         this.config.maxBufferHole,
         this.config.maxFragLookUpTolerance,
       );
       if (
-        delta > 0 &&
-        (delta < skipTolerance ||
-          (this.loadingParts && delta < 2 * (details?.partTarget || 0)))
+        (seekToSegmentBoundary && newStartPosition < startPosition) ||
+        (delta > 0 &&
```

Comment:
So couldn't this change be simplified to always adjust the start based on the delta which uses all buffered media?

```suggestion
        seekToSegmentBoundary ||
        (delta > 0 &&
```

Reply:
You had mentioned earlier in the feature request, that we should not adjust the start position to be later than the original start position. Because we were going to try to snap to both audio and video, this would have been necessary, but because we're only snapping to main, it should only be able to go to a smaller value. I've made this change now.